--- a/include/frg/printf.hpp
+++ b/include/frg/printf.hpp
@@ -361,10 +361,12 @@ frg::expected<format_error> printf_format(A agent, const char *s, va_struct *vs
 			switch(pos_args[i].type) {
 				case printf_arg_type::CHAR:
 					pop_arg<char>(vsp, &opts);
 					break;
+#ifdef _GLIBCXX_USE_WCHAR_T
 				case printf_arg_type::WCHAR:
 					pop_arg<wint_t>(vsp, &opts);
 					break;
+#endif
 				case printf_arg_type::INT:
 					switch (pos_args[i].sz) {
 						case printf_size_mod::char_size:
